external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -1445,22 +1458,29 @@ impl DependencyGraph {
                         format!("Parsing response from '{resolver}' for dependency '{to_name}' of package '{from_id}'")
                     })?;
 
-                    let mut root_sub_package = None;
-                    for (_from_package, to_pkg, _) in sub_graph.package_graph.edges(from_id) {
-                        if root_sub_package.is_some() {
-                            // TODO: We can in fact allow allow multiple root packages / graphs and relax this constraint.
-                            bail!("Multiple root packages in external dependencies found but expected only one");
-                        }
-                        root_sub_package = Some(to_pkg);
-                    }
-                    let Some(root_sub_package) = root_sub_package else {
-                        bail!(
-                            "Expected a root package in external dependencies group but none found"
-                        );
+                    let root_sub_package_id = match sub_graph
+                        .package_graph
+                        .edges(from_id)
+                        .collect::<Vec<_>>()
+                        .as_slice()
+                    {
+                        [(_, id, _)] => *id,
+                        // TODO: We can in fact allow allow multiple root packages / graphs and relax this constraint.
+                        _ => bail!("Expected a single root dependency but none or multiple found"),
```

Comment:
not a fan of `let ... else`? 😉

Reply:
`let ... else` is cool but had to do `.as_slice()` because it doesn't work on vector and then it returned `&Symbol` instead of `Symbol` so decided to go with `match` and deref in the arm so I don't have to do it later in 2 places.

Also gave me some borrow errors 
![image](https://github.com/user-attachments/assets/a942b4c5-0987-4e3f-8436-343720cdfcc7)

 I'm still not very proficient in Rust so these tips actually help a lot (I didn't know about `let ... else`)!

Reply:
Ah yes, I think you can work around these constraints like this (note the `&`):

```rust
let [(_, &root_sub_package_id, _)] = sub_graph
    .package_graph
    .edges(from_id)
    .collect::<Vec<_>>()
    .as_slice()
{
    bail!("...");
};
```

Reply:
Ah indeed this works but with `let &[(_, root_sub_package_id, _)]  = <...>`